pr_agent/algo/utils.py

```diff
@@ -82,3 +88,17 @@ def try_fix_json(review, max_iter=10):
             logging.error("Unable to decode JSON response from AI")
             data = {}
     return data
+
+def fix_json_escape_char(json_message=None):
+    result = None
+    try:
+        result = json.loads(json_message)
+    except Exception as e:
+        # Find the offending character index:
+        idx_to_replace = int(str(e).split(' ')[-1].replace(')', ''))
+        # Remove the offending character:
+        json_message = list(json_message)
+        json_message[idx_to_replace] = ' '
+        new_message = ''.join(json_message)
+        return fix_JSON(json_message=new_message)
```

Comment:
**Suggestion:** The function `fix_json_escape_char` is recursively calling itself which can lead to a maximum recursion depth error if the JSON message has many errors. Consider using a loop instead of recursion.
```suggestion
def fix_json_escape_char(json_message=None):
    while True:
        try:
            return json.loads(json_message)
        except Exception as e:
            # Find the offending character index:
            idx_to_replace = int(str(e).split(' ')[-1].replace(')', ''))
            # Remove the offending character:
            json_message = list(json_message)
            json_message[idx_to_replace] = ' '
            json_message = ''.join(json_message)
```